File: frontend/lib/model/host.dart
class Host{
 final int? id;
 final String? name;
 final String? ip;
 final int? port;
 final String? username;
 final String? password;
 final int? rubixPort;
 final String? rubixUsername;
 final String? rubixPassword;

  Host({ this.id,this.name,this.ip,this.port,this.username,this.password,this.rubixPort,this.rubixUsername,this.rubixPassword});
  factory Host.fromJson(Map<String,dynamic> json){
    return Host(
      id: json['id'] ,
      name: json['name'] ,
      ip: json['ip'] ,
      port: json['port'] ,
      username: json['username'] ,
      password: json['password'] ,
      rubixPort: json['rubix_port'] ,
      rubixUsername: json['rubix_username'] ,
      rubixPassword: json['rubix_password'] ,
    );
  }
  @override
  String toString() {
    // TODO: implement toString
    return 'Host{id: $id, name: $name}';
  }
}

File: backend/model/host.go
package model

import "time"

type Host struct {
	ID                   string    `json:"id" gorm:"primarykey"  get:"true" delete:"true"`
	Name                 string    `json:"name"  gorm:"type:varchar(255);unique;not null" required:"true" get:"true" post:"true" patch:"true"`
	IP                   string    `json:"IP" required:"true" default:"192.168.15.10" get:"true" post:"true" patch:"true"`
	Port                 int       `json:"port" required:"true" default:"22" get:"true" post:"true" patch:"true"`
	HTTPS                *bool     `json:"HTTPS" get:"true" post:"true" patch:"true"`
	Username             string    `json:"username" required:"true" default:"admin" get:"true" post:"true" patch:"true"`
	Password             string    `json:"password" required:"true" get:"false" post:"true" patch:"true"`
	RubixPort            int       `json:"rubix_port" required:"true" default:"1660" get:"true" post:"true" patch:"true"`
	RubixUsername        string    `json:"rubix_username" required:"true" default:"admin" get:"true" post:"true" patch:"true"`
	RubixPassword        string    `json:"rubix_password" required:"true" post:"true" patch:"true"`
	IsLocalhost          bool      `json:"is_localhost" get:"true" post:"true" patch:"true"`
	RubixToken           string    `json:"-"`
	RubixTokenLastUpdate time.Time `json:"-"`
}